doors_dashboards/core/featurehandler.py

```diff
@@ -145,20 +161,25 @@ def _read_features(self, features: Dict) -> pd.DataFrame:
             )
 
     def get_points_as_tuples(self, collection: str = None) -> \
-            Tuple[List[float], List[float], List[str]]:
+            Tuple[List[float], List[float], List[str], List[float], List[str]]:
         collection = self._selected_collection if not collection else collection
-        df = self.get_df(collection)
-        lons = list(df["lon"])
-        lats = list(df["lat"])
+        gdf = self.get_df(collection)
+
+        lons = list(gdf.geometry.apply(lambda p: p.x))
+        lats = list(gdf.geometry.apply(lambda p: p.y))
+        customdata = [collection] * len(lons)
```

Comment:
That is just the collection name. Why does it need to be returned here? Not sure this is the best approach,

Reply:
We need collection name in custom data because whenever the point is clicked we need  to know that the clicked point belongs to which collection since we have multiple collections on map.

Reply:
But then we only need the term collection once. Also, the method get_points_as_tuples is called with a collection as parameter, so we know this before we call the method. If we don't know, get_points_as_tuples is not the method we should ask. As a matter of fact, this method is still using the selected_collection bit which should be removed soon.